tests/test/org/commcare/test/utilities/MockApp.java

```diff
@@ -30,23 +33,37 @@ public MockApp(String resourcePath) throws Exception {
         if(!(resourcePath.startsWith("/") && resourcePath.endsWith("/"))) {
             throw new IllegalArgumentException("Invalid resource path for a mock app " + resourcePath);
         }
+        APP_BASE = resourcePath;
         LivePrototypeFactory mPrototypeFactory = setupStaticStorage();
         MockUserDataSandbox mSandbox = new MockUserDataSandbox(mPrototypeFactory);
         CommCareConfigEngine mEngine = new CommCareConfigEngine(mPrototypeFactory);
 
-        mEngine.installAppFromReference("jr://resource" + resourcePath + "profile.ccpr");
+        mEngine.installAppFromReference("jr://resource" + APP_BASE + "profile.ccpr");
         mEngine.initEnvironment();
-        ParseUtils.parseIntoSandbox(this.getClass().getResourceAsStream(resourcePath + "user_restore.xml"), mSandbox);
+        ParseUtils.parseIntoSandbox(this.getClass().getResourceAsStream(APP_BASE + "user_restore.xml"), mSandbox);
 
         //If we parsed in a user, arbitrarily log one in.
-        for(IStorageIterator<User> iterator = mSandbox.getUserStorage().iterate(); iterator.hasMore();) {
-            mSandbox.setLoggedInUser(iterator.nextRecord());
-            break;
-        }
+        mSandbox.setLoggedInUser(mSandbox.getUserStorage().read(0));
 
         mSessionWrapper = new SessionWrapper(mEngine.getPlatform(), mSandbox);
     }
 
+    /**
+     * Loads the provided form and properly initializes external data instances,
+     * such as the casedb and commcare session.
+     */
+    public FormParseInit loadAndInitForm(String formFileInApp) {
```

Comment:
Not used now that I found a minimal test case, but it was helpful in copying a form and user restore from HQ and loading all that data into a `commcare-core` level test. Leaving it here in hopes that we use it soon. Also willing to delete it if that sounds better.